lit_gpt/tokenizer.py

```diff
@@ -25,16 +28,19 @@ def __init__(self, checkpoint_dir: Path) -> None:
                 with open(special_tokens_path) as fp:
                     config = json.load(fp)
                 self.bos_id = config.get("bos_token_id")
+                self.use_bos = self.bos_id  is not None
```

Comment:
@carmocca wasn't sure if I should also change this to apply `check_if_bos_token_used`. My thinking is that if the special tokens path has a bos_token id, we should assume it is used

Reply:
I would call it, the bos_token_id might be defined even if it's not used.

Reply:
Updated this to also use the same check, marking this as resolved.